httpclient/clientBuilder.go

```diff
@@ -33,13 +45,27 @@ func (builder *httpClientBuilder) Build() (*HttpClient, error) {
 	if builder.certificatesDirPath == "" {
 		transport := createDefaultHttpTransport()
 		transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: builder.insecureTls}
+		if builder.clientCertificatePath != "" {
+			cert, err := tls.LoadX509KeyPair(builder.clientCertificatePath, builder.clientCertificateKeyPath)
+			if err != nil {
+				return nil, errorutils.CheckError(errors.New("Failed creating HttpClient: " + err.Error()))
+			}
+			transport.TLSClientConfig.Certificates = []tls.Certificate{cert}
+		}
 		return &HttpClient{Client: &http.Client{Transport: transport}}, nil
 	}
 
 	transport, err := cert.GetTransportWithLoadedCert(builder.certificatesDirPath, builder.insecureTls, createDefaultHttpTransport())
 	if err != nil {
 		return nil, errorutils.CheckError(errors.New("Failed creating HttpClient: " + err.Error()))
 	}
+	if builder.clientCertificatePath != "" {
+		cert, err := tls.LoadX509KeyPair(builder.clientCertificatePath, builder.clientCertificateKeyPath)
+		if err != nil {
+			return nil, errorutils.CheckError(errors.New("Failed creating HttpClient: " + err.Error()))
+		}
+		transport.TLSClientConfig.Certificates = []tls.Certificate{cert}
+	}
```

Comment:
The above code is duplicated inside this method.
Let's take the repeating code into a different method and reuse it.